rdmo/management/tests/test_frontend.py

```diff
@@ -0,0 +1,213 @@
+import os
+import re
+from dataclasses import dataclass
+from urllib.parse import urlparse
+
+import pytest
+
+from django.core.management import call_command
+
+from playwright.sync_api import Page, expect
+from pytest_django.live_server_helper import LiveServer
+
+from rdmo.accounts.utils import set_group_permissions
+from rdmo.conditions.models import Condition
+from rdmo.core.models import Model
+from rdmo.domain.models import Attribute
+from rdmo.options.models import Option, OptionSet
+from rdmo.questions.models import Catalog, Question, Section
+from rdmo.questions.models import Page as PageModel
+from rdmo.questions.models.questionset import QuestionSet
+from rdmo.tasks.models import Task
+from rdmo.views.models import View
+
+pytestmark = pytest.mark.e2e
+
+# needed for playwright to run
+os.environ.setdefault("DJANGO_ALLOW_ASYNC_UNSAFE", "true")
+
+
+@dataclass
+class ModelHelper:
+    """Helper class to bundle information about models for test cases."""
+
+    model: Model
+    form_field: str = "URI Path"
+    db_field: str = "uri_path"
+    has_nested: bool = False
+
+    @property
+    def url(self) -> str:
+        return f"{self.model._meta.model_name}s"
+
+    @property
+    def url_name(self) -> str:
+        """Return the url name for the model."""
+        url = self.url.rstrip("s")
+        url = url.replace("set", " set")
+        return url
+
+    @property
+    def verbose_name(self) -> str:
+        """Return the verbose_name for the model."""
+        return self.model._meta.verbose_name
+
+    @property
+    def verbose_name_plural(self) -> str:
+        """Return the verbose_name_plural for the model."""
+        return self.model._meta.verbose_name_plural
+
+
+@pytest.fixture(scope="function")
+def e2e_tests_django_db_setup(django_db_setup, django_db_blocker, fixtures):
+    """Set up database and populate with fixtures, that get restored for every test case."""
+    with django_db_blocker.unblock():
+        call_command("loaddata", *fixtures)
+        set_group_permissions()
+
+
+@pytest.fixture(scope="session")
+def base_url(live_server: LiveServer) -> str:
+    """Enable playwright to address URLs with base URL automatically prefixed."""
+    return live_server.url
+
+
+@pytest.fixture
+def logged_in_admin_user(e2e_tests_django_db_setup, page: Page) -> Page:
+    """Log in as admin user through django login UI, returns logged in page for e2e tests."""
+    page.goto("/account/login")
+    page.get_by_label("Username").fill("admin", timeout=5000)
+    page.get_by_label("Password").fill("admin")
+    page.get_by_role("button", name="Login").click()
+    page.goto("/management")
+    yield page
+
+
+def test_management_navigation(logged_in_admin_user: Page) -> None:
```

Comment:
thought to make a for-loop of `model_helper` inside this test function, that could use the `verbose_name_plural` as value for eg. `name="Question sets"` , it might need a `.capitalize` to get `Questions sets` instead of `Question Sets`

Reply:
Is this wrong?

https://github.com/rdmorganiser/rdmo/blob/a54867e9604bf7ed604ff45ce219bbc33f5ff21e/rdmo/questions/models/questionset.py#L188

Reply:
yes that looks like a bug 😉 
In the models there is actually also a `verbose_name` and `_plural` property that include the translation as well..
https://github.com/rdmorganiser/rdmo/blob/a54867e9604bf7ed604ff45ce219bbc33f5ff21e/rdmo/questions/models/questionset.py#L205
then those could be used directly instead of the properties on the `ModelHelper` 🙈

Reply:
Alright. The Modelhelper itself is still useful, right?
I will remove the properties and call the model.verbose_name stuff directly.

Also I added a fix PR for the questionset verbose name in #817

Reply:
What do you think @MyPyDavid? What info should remain in the modelhelper class?

Reply:
think its fine like that, the properties on the modelhelper as "shortcuts" make it more readable anyways